subsystem-test/framework/src/main/java/org/jboss/as/subsystem/test/TestParser.java

```diff
@@ -77,7 +78,11 @@ public void writeContent(XMLExtendedStreamWriter writer, ModelMarshallingContext
         } finally {
             writer.setDefaultNamespace(defaultNamespace);
         }
+        if (empty){
+            writer.writeEmptyElement(Element.SUBSYSTEM.getLocalName());
+        }
```

Comment:
I've changed it to be in try block, as namespace is not relevant in this scenario at all.